### DISCUSSION TITLE
Regarding the Pull and Push Management in GDS

### DISCUSSION THREAD

**Discussion by RakshanPremsagar** [eclipse-milo/milo#787] [General]
Hello Kevin,

As per the LDS or multicast subnet discovery, the OPC UA server needs to be registered to the LDS using the services provided by the discoveryServiceSet( RegisterServer2/RegisterServer)

Now moving on to the GDS, it encorporates the Certificate management along with discovery functionality. 
Considering only the discovery functionality(and ignoring the certificate management), is the mechanism right to say that we can register(RegisterServer2/RegisterServer) the OPC UA server to the GDS and then call the queryServers method? I believe the discoveryServiceSet cannot be used here, as the specification says the use of RegisterApplication. If so, is there any reason behind it?

Also, could you please suggest me an example for pull and push management? I failed to understand what the specification says to be honest...
Eg: If i have 2 hosts A and B. A has the GDS running, B has the GDS client running and it is connected to the GDS server.
Does the registration of the OPC UA(extra server) server running on A to the GDS happen after the GDS client connects to the GDS Server? Or is it a thing wherein, the GDS client needs to take the initiative to register the application using XML files to the GDS Server?(Ref: [GDS demo](https://www.youtube.com/watch?v=TCy8JlnWIXw)) And how do we configure these XML files? 
Been a lot unclear with the Pull and Push management mechanism in GDS.

Sorry too many questions I believe. But, really looking forward for some answers here.

Regards,
Rakshan

**Comment by kevinherron** [comment]
Try asking here: https://opcfoundation.org/forum/

You seem to have a lot of Discovery related questions and as I said during one of our first interactions, I'm not really a Discovery expert :)

Try to get Randy Armstrong's attention.

The only experience with GDS I've had was at an interop event where we tested a GDS server pushing certificate data into the Eclipse Milo Demo Server. I think the means of registering information about the server with the GDS in the first place so it knew where to push happened out of band via a web page specific to that GDS implementation, but I might be mis-remembering.

**Reply by RakshanPremsagar** [reply]
Thanks Kevin, been active on the OPC foundation too
